packages/esm-form-engine-app/src/form-collapse-toggle/form-collapse-toggle.component.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Toggle } from '@carbon/react';
+import styles from './styles.scss';
+
+const FormCollapseToggle = () => {
+  const { t } = useTranslation();
+  const [isFormEmbedded, setIsFormEmbedded] = useState<boolean>(false);
+
+  useEffect(() => {
+    const handleFormEmbedded = (event) => {
+      setIsFormEmbedded(event?.detail?.value || false);
+    };
+
+    window.addEventListener('openmrs:form-view-embedded', handleFormEmbedded);
+
+    return () => {
+      window.removeEventListener('openmrs:form-view-embedded', handleFormEmbedded);
+    };
+  }, []);
+
+  const handleOnToggle = (value: boolean) => {
```

Comment:
This should be wrapped in `useCallback()` or hoisted out of the React component.